pkg/adapter/adapter.go

```diff
@@ -355,6 +395,84 @@ func (a *Adapter) postMessage(msg wabbit.Delivery) error {
 	return nil
 }
 
+func isBinary(headers wabbit.Option) bool {
+	return headers["ce-specversion"] != nil || headers["Ce-Specversion"] != nil
+}
+
+func isStructured(body map[string]interface{}) bool {
+	return body["specversion"] != nil || body["Specversion"] != nil
+}
+
+func setBinaryMessageProperties(event *cloudevents.Event, headers wabbit.Option) error {
+	for ceKey, val := range headers {
+		lowerKey := strings.ToLower(ceKey)
+
+		if lowerKey == "content-type" && val != nil {
+			lowerKey = "ce-datacontenttype"
+		}
+
+		if strings.HasPrefix(lowerKey, prefix) {
+			key := strings.TrimPrefix(lowerKey, prefix)
+			err := setEventAttributes(event, key, lowerKey, val)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func setStructuredMessageProperties(event *cloudevents.Event, body map[string]interface{}) error {
+	for key, val := range body {
```

Comment:
you need to handle the key `data` special, it becomes the event data. I wonder if there's a sdk-go library function that does this whole thing for us?

Reply:
I think for that this has to become an implementation of a protocol binding right?